cmd/dockerd/daemon.go

```diff
@@ -287,24 +239,39 @@ func (cli *DaemonCli) start() (err error) {
 		"graphdriver": d.GraphDriverName(),
 	}).Info("Docker daemon")
 
-	cli.initMiddlewares(api, serverConfig)
-	initRouter(api, d, c)
+	// setup the default routes
+	defaultRoutes := getRouters(d, c)
+	api.InitRouter(utils.IsDebugEnabled(), defaultRoutes...)
+
+	// setup the introspections routes without cluster capabilities
```

Comment:
It's always a backdoor. You can even mount the original docker socket using the "instrospection" socket.